crates/sargon/src/core/types/intent_discriminator.rs

```diff
@@ -0,0 +1,157 @@
+pub use crate::prelude::*;
+
+// Generate the FfiConverter needed by UniFFI for newtype `IntentDiscriminator`.
+uniffi::custom_newtype!(IntentDiscriminatorSecretMagic, u64);
+
+#[derive(
+    Clone, Copy, PartialEq, Eq, Hash, derive_more::Display, derive_more::Debug,
+)]
+pub struct IntentDiscriminatorSecretMagic(pub u64);
```

Comment:
what is an intent discriminator? Missing doc :) and I do not understand from the type name.